### PR TITLE
Significantly reduced warnings 

### PR DESCRIPTION
**Pull request by trooper2013**
There were 60+ warnings when building "everything".  Now the number is reduced to 34 of which  at least 30 or so warnings are related to code in third party libs.

1. Modified code that used deprecated UIAlert & UIActionSheet.
2. Added images and removed images that aren't quite required for iOS 8.0+ (Unassigned images lead to significant number of warnings in xcode8)
3. Fixed launch screen from use of storyboard to image assets. 
4. Added flags to indicate extension safe usage for libs(s) and the opposite for tests.

### CODE REVIEW

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -76,7 +77,7 @@
 		BCC8F2271D46C01900171499 /* HostPage.swift in Sources */ = {isa = PBXBuildFile; fileRef = BCC8F2261D46C01900171499 /* HostPage.swift */; };
 		CE2B8D671CE932D600C6FC6A /* SalesforceAnalytics.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = CE2B8D4C1CE932BD00C6FC6A /* SalesforceAnalytics.framework */; };
 		CE6AB39B1C10C4F50012125E /* libz.tbd in Frameworks */ = {isa = PBXBuildFile; fileRef = CE6AB39A1C10C4F50012125E /* libz.tbd */; };
-		CEF4A2EE1D9491010025EDF0 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = CEF4A2EC1D9491010025EDF0 /* LaunchScreen.storyboard */; };
+		CEF4A2EE1D9491010025EDF0 /* (null) in Resources */ = {isa = PBXBuildFile; };
```
Comment (khawkins) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj]: 1. Go to SmartSyncExplorer project settings.
2. Select the SmartSyncExplorer target.
3. Select Build Phases.
4. Expand Copy Bundle Resources.
5. Remove the empty entry in that list.

Voila.  Null project item goes away. 🙂

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/ContactListViewController.m [hunk 2]
```diff
@@ -613,20 +631,20 @@ - (void)clearPopovers:(NSNotification *)note
...
-    if ([actionSheet isEqual:self.logoutActionSheet]) {
-        self.logoutActionSheet = nil;
-        if (buttonIndex == actionSheet.destructiveButtonIndex) {
-            [[SFAuthenticationManager sharedManager] logout];
-        }
+        [self.logoutActionSheet dismissViewControllerAnimated:YES completion:nil];
     }
 }
+//
+//#pragma mark - UIActionSheetDelegate
```
Comment (khawkins) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/ContactListViewController.m]: Should we just remove this block?

**Comment by khawkins** [comment]
Can the RestAPIExplorer launch screen assets be rolled into the shared assets?

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -944,6 +947,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CE2B8CAB1CE9281F00C6FC6A /* SalesforceAnalytics-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: Should probably be `YES` for library projects.

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -3380,6 +3382,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CEBA94EF1C1B5D4C00C0B2E0 /* SalesforceSDKCore-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Should be `YES` here.

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -3421,6 +3424,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CEBA94FE1C1B990A00C0B2E0 /* SalesforceSDKCore-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Should be `YES` here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1729,6 +1735,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CECB662E1C1BB8CB008038AE /* SalesforceHybridSDK-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: Should be `YES` here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -1768,6 +1775,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CECB662F1C1BB8CB008038AE /* SalesforceHybridSDK-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: Should be `YES` here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1650,6 +1653,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: Should be `YES` here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -1610,6 +1612,7 @@
...
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: Should be `YES` here.

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -920,6 +922,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CE2B8CAA1CE9281F00C6FC6A /* SalesforceAnalytics-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: Should be `YES` here.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 11]
```diff
@@ -1,95 +1,43 @@
...
+      "subtype" : "736h",
+      "scale" : "3x"
     },
     {
       "orientation" : "portrait",
       "idiom" : "ipad",
       "filename" : "Default-Portrait.png",
       "extent" : "full-screen",
-      "minimum-system-version" : "9.0",
+      "minimum-system-version" : "7.0",
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 12]
```diff
@@ -1,95 +1,43 @@
...
-    },
-    {
       "extent" : "full-screen",
-      "idiom" : "iphone",
-      "subtype" : "retina4",
-      "filename" : "Default-568h@2x.png",
-      "minimum-system-version" : "9.0",
-      "orientation" : "portrait",
-      "scale" : "2x"
+      "minimum-system-version" : "8.0",
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 13]
```diff
@@ -1,95 +1,43 @@
...
       "scale" : "1x"
     },
     {
       "orientation" : "landscape",
       "idiom" : "ipad",
-      "filename" : "Default-Landscape.png",
+      "filename" : "Default1024x768.png",
       "extent" : "full-screen",
-      "minimum-system-version" : "9.0",
+      "minimum-system-version" : "7.0",
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 14]
```diff
@@ -1,95 +1,43 @@
...
       "scale" : "1x"
     },
     {
       "orientation" : "portrait",
       "idiom" : "ipad",
-      "filename" : "Default-Portrait@2x.png",
+      "filename" : "Default1536x2048.png",
       "extent" : "full-screen",
-      "minimum-system-version" : "9.0",
+      "minimum-system-version" : "7.0",
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 15]
```diff
@@ -1,95 +1,43 @@
...
-      "orientation" : "portrait",
-      "idiom" : "ipad",
-      "extent" : "to-status-bar",
-      "scale" : "2x"
-    },
-    {
-      "orientation" : "landscape",
-      "idiom" : "ipad",
-      "extent" : "to-status-bar",
+      "minimum-system-version" : "7.0",
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj [hunk 16]
```diff
@@ -1592,8 +1602,8 @@
 		827C634819E6018C00027484 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj]: Should be `YES` here.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj [hunk 17]
```diff
@@ -1575,8 +1585,8 @@
 		827C634719E6018C00027484 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj]: Should be `YES` here.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/ContactListViewController.m [hunk 18]
```diff
@@ -430,12 +430,14 @@ - (void)popoverOptionSelected:(NSString *)text {
...
     if ([text isEqualToString:kActionLogout]) {
-        self.logoutActionSheet = [[UIActionSheet alloc] initWithTitle:@"Are you sure you want to log out?"
-                                                             delegate:self
-                                                    cancelButtonTitle:@"Cancel"
-                                               destructiveButtonTitle:@"Confirm Logout"
-                                                    otherButtonTitles:nil];
-        [self.logoutActionSheet showFromBarButtonItem:self.moreButton animated:YES];
+        [self showLogoutActionSheet];
+
+//                [[UIActionSheet alloc] initWithTitle:@"Are you sure you want to log out?"
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/ContactListViewController.m]: Remove commented out code.

File: shared/resources/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 19]
```diff
@@ -1,11 +1,17 @@
...
+      "extent" : "full-screen",
+      "scale" : "1x"
+    },
+    {
       "extent" : "full-screen",
       "idiom" : "iphone",
       "subtype" : "736h",
       "filename" : "Default1242x2208.png",
-      "minimum-system-version" : "9.0",
+      "minimum-system-version" : "8.0",
```
Comment (bhariharan) [inline shared/resources/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: shared/resources/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 20]
```diff
@@ -14,82 +20,17 @@
       "idiom" : "iphone",
       "subtype" : "736h",
       "filename" : "Default2208x1242.png",
-      "minimum-system-version" : "9.0",
+      "minimum-system-version" : "8.0",
```
Comment (bhariharan) [inline shared/resources/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: shared/resources/Images.xcassets/LaunchImage.launchimage/Contents.json [hunk 21]
```diff
@@ -14,82 +20,17 @@
...
-    {
-      "orientation" : "landscape",
-      "idiom" : "ipad",
-      "extent" : "to-status-bar",
-      "scale" : "1x"
-    },
-    {
+      "subtype" : "667h",
+      "filename" : "Default750x1334.png",
+      "minimum-system-version" : "8.0",
```
Comment (bhariharan) [inline shared/resources/Images.xcassets/LaunchImage.launchimage/Contents.json]: Should be `9.0`.

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj (hunk 10 above)
Comment (trooper2013) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: After a close look at it the entry is related to SalesforceAnalyticsTestApp. Shouldn't be marked as YES.

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj (hunk 3 above)
Comment (trooper2013) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: SalesforceAnalyticsTestApp entry again.

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj (hunk 4 above)
Comment (trooper2013) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Seems to be test app related. Should be NO.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj (hunk 17 above)
Comment (trooper2013) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj]: SmartSyncExplorer App related. Should be NO.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj (hunk 16 above)
Comment (trooper2013) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj]: SmartSyncExplorerApp related. Must be NO here too.

**Comment by khawkins** [comment]
I presume we want to be careful with setting `APPLICATION_EXTENSION_API_ONLY` to `YES`, in general.  According to [the docs](https://pewpewthespells.com/blog/buildsettings.html#application_extension_api_only):
> When enabled, this causes the compiler and linker to disallow use of APIs that are not available to app extensions and to disallow linking to frameworks that have not been built with this setting enabled. If enabled, passes flag -fapplication-extension.

That sounds fairly restrictive.  But I suppose we could try it out....

**Comment by khawkins** [comment]
The RestAPIExplorer launch images were retained, independently of the shared launch images.  Can we remove those and link to the shared images?

**Pull request merged**